tests/chainer_tests/test_function.py

```diff
@@ -405,7 +405,7 @@ class FunctionWithRetaining(chainer.Function):
 
     def forward(self, inputs):
         self.retain_inputs([1])
-        self.retain_outputs([1])
+        self.retain_outputs([1], retain_after_backward=True)
```

Comment:
Please make tests for botch `retain_after_backward=True` and `False`, and check if `output_data` is removed or not.